interfaces/dbus/backend.go

```diff
@@ -132,6 +156,70 @@ func addContent(securityTag string, executableSnippets [][]byte, content map[str
 	}
 }
 
+func (b *Backend) setupBusServ(snapInfo *snap.Info, snippets map[string][][]byte) error {
+	snapName := snapInfo.Name()
+
+	content := map[string]*osutil.FileState{}
+	for _, appInfo := range snapInfo.Apps {
+		securityTag := appInfo.SecurityTag()
+
+		// FIXME: layer violation, we really should do that in the
+		//        dbus interface but without PR #2613 its really tricky
+		for _, slot := range appInfo.Slots {
+			if slot.Interface != "dbus" {
+				continue
+			}
+			bus, ok := slot.Attrs["bus"].(string)
+			if !ok {
+				continue
+			}
+			if bus != "session" {
+				continue
+			}
+			name, ok := slot.Attrs["name"].(string)
+			if !ok {
+				continue
+			}
+			// we check if its a service here so that we know
+			// if a dbus service file needs to get generated.
+			isService, ok := slot.Attrs["service"].(bool)
+			if !ok || !isService {
+				continue
+			}
+
+			var buffer bytes.Buffer
```

Comment:
Can you add a comment here. Eg:
```
// We set only 'Name' and 'Exec' for now. We may add 'User' for 'system'
// services when we support per-snap users. Don't specify 'SystemdService'
// and just let dbus-daemon launch the service since 'SystemdService' is only
// used by dbus-daemon to tell systemd to launch the service and systemd
// user sessions aren't available everywhere yet.